In [1]:
import pandas as pd
import numpy as np

In [2]:
customer = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_customer.csv", sep = ";")
accounts = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_accounts.csv", sep = ";")
transactions = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_transactions.csv", sep = ";")

account_checking = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_checking_acct.csv", sep = ";")
account_checking_transactions = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_checking_tran.csv", sep = ";")

saving_account = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_savings_acct.csv", sep = ";")
saving_transactions = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_savings_tran.csv", sep = ";")

credit_account = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_credit_acct.csv", sep = ";")
credit_transactions = pd.read_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\original\\twm_credit_tran.csv", sep = ";")

In [3]:
# Remove the white space from string columns
all_data = [customer, accounts, account_checking, account_checking_transactions, saving_account, saving_transactions, credit_account, credit_transactions]
for i in range(len(all_data)):
    for white_space in all_data[i].select_dtypes('object'):
        all_data[i][white_space] = all_data[i][white_space].str.strip()
customer, accounts, account_checking, account_checking_transactions, saving_account, saving_transactions, credit_account, credit_transactions = all_data

In [4]:
# Make account number uniform as string datatype
account_number = [accounts, transactions, account_checking, saving_account, credit_account]
for number in account_number:
    number['acct_nbr'] = number['acct_nbr'].astype(str)
accounts, transactions, account_checking, saving_account, credit_account = account_number

In [5]:
#transactions_data = [transactions, account_checking_transactions, saving_transactions, credit_transactions]
#for i in range(len(transactions_data)):
#    transactions_data[i]['tran_time'] = transactions_data[i]['tran_time'].astype(str)
#    transactions_data[i]['tran_time'] = transactions_data[i]['tran_time'].replace('',np.nan)
#    transactions_data[i].loc[(transactions_data[i]['tran_time'].str.len() > 6) | (transactions_data[i]['tran_time'].str.len() <=4),'tran_time'] = np.nan
#    transactions_data[i]['tran_time'] = pd.to_datetime(transactions_data[i]['tran_time'], format='%H%M%S').dt.hour

In [6]:
customer.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\customer.csv", index = False)

accounts.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\accounts.csv", index = False)

transactions.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\transactions.csv", index = False)

account_checking.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\account_checking.csv", index = False)
account_checking_transactions.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\account_checking_transactions.csv", index = False)

saving_account.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\saving_accounts.csv", index = False)
saving_transactions.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\saving_transactions.csv", index = False)

credit_account.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\credit_account.csv", index = False)
credit_transactions.to_csv("D:\\Python(New)\\Project\\Project_Week_5\\Mini_Project_Week_5\\clean\\credit_transactions.csv", index = False)

In [11]:
accounts[['acct_type','account_active']].value_counts()

acct_type  account_active
CK         Y                 468
CC         Y                 423
SV         Y                 375
CK         N                  52
SV         N                  46
CC         N                  45
dtype: int64

We have 375 active saving accounts and 423 credit accounts in total of 1266 active accounts